course_dashboard/answers_distribution.py

```diff
@@ -0,0 +1,178 @@
+            great_grandparent = store.get_parent_location(grandparent)
+            if great_grandparent:
+                ancestors_names['great_grandparent'] = store.get_item(great_grandparent).display_name
+    problem_module.ancestors_names = ancestors_names
+    return 
+
+def parse_problem_data_from_problem_module(problem_module):
+        problem_module_as_xml = ET.fromstring((problem_module.data).encode('utf-8'))
+        problem_module.as_xml = problem_module_as_xml
+
+def add_answers_distribution_to_problem_module(problem_module):
+    """Return problem module as xml tree with answers distibution stats"""
+
+    problem_module.as_xml = ET.fromstring((problem_module.data).encode('utf-8'))
+    problem_position_in_module = 0
+
+    for element in problem_module.as_xml.iter():
+        if element.tag in problem_handlers:
+            problem_position_in_module += 1
+            problem_handlers[element.tag](problem_module, element, problem_position_in_module)
+
+def checkbox_handler(problem, problem_module, problem_number):
+    """Handler for checkbox problem"""
+    return
+
+def text_input_handler(problem, problem_module, problem_number):
+    """Handler for text input problem """
+    return
+
+def not_handled(problem, problem_module, problem_number):
+    """Problem not handled yet"""
+    return
+    
+def multiplechoice_handler(problem_module, problem, problem_position):
+    """
+    handle multiplechoice problem
+
+       problem :
+       <multiplechoiceresponse>
+         <choicegroup label="What Apple device competed with the portable CD player?" type="MultipleChoice">
+           <choice correct="false">The iPad</choice>
+           <choice correct="false">Napster</choice>
+           <choice correct="true">The iPod</choice>
+           <choice correct="false">The vegetable peeler</choice>
+         </choicegroup>
+       </multiplechoiceresponse>
+
+       Only one answer is possible, so we create a 'total_response' attribute for each choice 
+       and increment it according to the student answer.
+    """
+
+    total_response = 0
+   
+    problem_module_id = problem_module.scope_ids.usage_id.name
+    problem_module_root_id_dashed = "i4x-{}-{}-problem-".format(problem_module.scope_ids.usage_id.org,
+                                                                problem_module.scope_ids.usage_id.course)
+
+    problem_module_results = get_student_responses_to_problem_module(problem_module)
+
+    for student_problem_module_results in problem_module_results:
+        # state field contains all answers of a problem module as json
+        state = json.loads(student_problem_module_results.state)
+        try:
+            answers = state["student_answers"]
+            if answers == {}: ## happen when the student submited his response but didn't answerd 
+                continue
+        except KeyError :
+            # happen when a student has loaded the problem but not yet answerd
+            continue
```

Comment:
More concisely:

    answers = state.get("student_answers", {})
    if not answers:
        continue